core/vm/instructions.go

```diff
@@ -814,7 +814,7 @@ func opCall(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byt
 		}
 	}
 
-	if interpreter.evm.chainRules.IsEIP4762 && !chargeCallVariantEIP4762(interpreter.evm, scope) {
+	if interpreter.evm.chainRules.IsEIP4762 && !transfersValue && !chargeCallVariantEIP4762(interpreter.evm, scope) {
```

Comment:
This is something that surfaced when I was helping Luis with some work on Besu. I've chatted about this with some people, but re-explaining here for completeness.

Imagine we have a value-bearing CALL (it can be for an existing address, so nothing weird). In this case, we'll execute L811 first which is adding the "value transfer" part to the witness -- this includes doing a write-event on the `target` `BASIC_DATA`.

Then in L817 we **were** (i.e: before this PR) doing the usuasl `*CALL` touch on `BASIC_DATA` for the target. In this case, we'd be charging 100 of gas... since the "value transfer" touching we did before already added `BASIC_DATA` to the witness -- so a "warm charge" is done.

This looks wrong? This extra 100 of gas doesn't make much sense, in the sense that could be easily removed (as I do in this PR).

Reply:
hmmm yeah we should at least merge the two checks for eip4762 into a single `if`, and have an `else` to call `chargeCallVariantEIP4762`.

And adding a comment would be nice, so that we remember it in 2 weeks: just explain why we don't charge the first 100 in this case.

Reply:
BTW, I think there is a bug here:what if we call the same address with values multiple times? the 2nd time, `TouchAndChargeValueTransfer` should charge 100 gas. I think the best way to do this would be to change `TouchAndChargeValueTransfer` to return `100` if nothing gets added to the witness.

Reply:
Agree! Apparently this is the only place we use `TouchAndChargeValueTransfer` so looks safe to solve it that way. I'll make the change.